types/src/contract_event.rs

```diff
@@ -148,17 +148,19 @@ impl EventWithProof {
     pub fn verify(
         &self,
         ledger_info: &LedgerInfo,
-        access_path: &AccessPath,
+        _access_path: &AccessPath,
         sequence_number: u64,
         transaction_version: Version,
         event_index: u64,
     ) -> Result<()> {
-        ensure!(
-            self.event.access_path() == access_path,
-            "Access path ({}) not expected ({}).",
-            self.event.access_path(),
-            *access_path,
-        );
+        //      TODO: With the new Event design, we can no longer link this query access path to the
```

Comment:
I'd rather resolve this or at least have an idea how to address this before landing.

A straightforward way is to always carry the latest AccountStateWithProof with the response. 
Another one is to let Event itself carry information about it's type so that we can check it against what's asked by the request. (need a mapping between access_path/type_of_event in the request and the type info on the Event.)

people, thoughts? @kphfb @sblackshear

Reply:
I agree that it makes sense to either solve this now or have a concrete plan for resolution. 

At a minimum, this function needs to take an event handle ID as input and compare it to the event handle ID in `self.event`. The question of how the client can supply that handle ID is a tougher one. Your first suggestion is certainly one way: you can supply an `AccountStateWithProof` + an access path, then read the handle ID from the account state using that access path.

We could also ask the client to remember the event handle IDs for each event that they care about. They will need to know this info for handles that will not always be stored at the same access path.